system-design/margo-overview/software-composition.md

```diff
@@ -0,0 +1,440 @@
+# Software Composition
+
+Applications can be found in completely different stages:
+
+1. "Application Packaging": application has been prepared and made ready for provisioning.
+2. "Application Provisioning": application is set up, configured, and made available for use to the device, but has not yet been deployed (started) to be used.
```

Comment:
I feel the line between "application provisioning" and "application deployment" isn't clear. If an application is made available at "application provisioning" stage, there's no stopping of an user accessing it, hence it's already "deployed"?

Reply:
You are right, "made available for use" is misleading. Probably "made available to be started" would be clearer, right?

Reply:
After some quick research, I've confirmed that you are right. "staged" is a better term than "provisioned". I will update it in the document. Thanks for your feedback!